k-distribution/k-tutorial/1_basic/02_basics/README.md

```diff
@@ -1 +1,212 @@
+square brackets after any sentence, including both productions and rules.
+Various attributes with built-in meanings exist in K and will be discussed
+throughout the tutorial, but the user can also give any sentence a user-defined
+attribute using the same syntax by simply using an identifier for the attribute
+which is not reserved by K.
+
+## Rules, Matching, and Variables
+
+Functions in K are given definitions using rules. A rule begins with the `rule`
+keyword and contains at least one **rewrite operator**. The rewrite operator
+is represented by the syntax `=>`. The rewrite operator is one of the built-in
+productions in K, and we will discuss in more detail how it can be used in
+future lessons, but for now, you can think of a rule as consisting of a 
+**left-hand side** and a **right-hand side**, separated by the rewrite
+operator. On the left-hand side is the name of the function and zero or more
+**patterns**. On the right-hand side is another pattern. The meaning of the
+rule is relatively simple, having defined these components. If the function
+is called with arguments that **match** the patterns on the left-hand side,
+then the return value of the function is the pattern on the right-hand side.
+
+For example, in the above example, if the argument of the `colorOf` function
+is `Banana()`, then the return value of the function is `Yellow()`.
+
+So far we have introduced that a constructor is a type of pattern in K. We
+will introduce more complex patterns in later lessons, but there is one other
+type of basic pattern: the **variable**. A variable, syntactically, consists
+of an uppercase identifier. However, unlike a constructor, a variable will
+match **any** pattern with one exception: Two variables with the same name
+must match the same pattern.
+
+Here is a more complex example:
+
+```k
+module LESSON-02-D
+
+  syntax Container ::= Jar(Fruit)
+  syntax Fruit ::= Apple() | Pear()
+
+  syntax Fruit ::= contentsOfJar(Container) [function]
+
+  rule contentsOfJar(Jar(F)) => F
+
+endmodule
+```
+
+In this example, `F` is a variable. It will match both `Apple()` and `Pear()`.
+The return value of the function is created by **substituting** the matched
+values of all of the variables into the variables on the right-hand side of
+the rule.
+
+To demonstrate, compile this definition and execute the following program with
+krun:
+
+```
+contentsOfJar(Jar(Apple()))
+```
+
+You will see when you run it that the program returns `Apple()`, because that
+is the pattern that was matched to `F`.
+
+## Exercises
+
+1. Extend the definition in lesson-02-a.k with the addition of blackberries
+   and kiwis. For simplicity, blackberries are black and kiwis are green. Then
+   compile your definition and test that your additional fruits are correctly
+   handled by the `colorOf` function.
+2. Create a new definition which defines an outfit as a hat, shirt, pants, and
+   shoes. Define a new sort, `Boolean`, with two constructors, true and false.
```

Comment:
I think this is a big leap to make given the explanation above. At least say something to the effect of "An outfit will have to be a *multi-argument constuctor* containing a ...", to make sure they look at the correct parts of the explanation.